hive/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -102,7 +102,7 @@ private TServer newThriftServer(TServerSocket socket, HiveConf conf) throws Exce
         .transportFactory(new TTransportFactory())
         .protocolFactory(new TBinaryProtocol.Factory())
         .minWorkerThreads(3)
-        .maxWorkerThreads(5);
+        .maxWorkerThreads(10);
```

Comment:
Would it make sense to expose this as a config option in `TestHiveMetastore`? A small number of worker threads helped us to catch a few connection leaks early on.

Reply:
Agree to make it configurable. 

The small threads number is OK for me as long as it is enough for new tests. So I'd like to know if we can release the thread/connection when each unit test finish?

Reply:
I think that using Spark's catalog will avoid some of these issues because tests will reuse the same connection.

Reply:
Make sense and the unit test passed when I change back to use Spark's catalog.

Reply:
Can you revert this change? I'd like to see if it works with the recent changes to reduce the number of clients in tests.

Reply:
Done.